bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/2D/Intersection.cpp

```diff
@@ -0,0 +1,96 @@
+
+    using ostk::math::geom::d2::Object ;
+    using ostk::math::geom::d2::objects::Point ;
+    using ostk::math::geom::d2::objects::PointSet ;
+    using ostk::math::geom::d2::objects::Line ;
+    using ostk::math::geom::d2::objects::Segment ;
+    using ostk::math::geom::d2::objects::LineString ;
+    using ostk::math::geom::d2::objects::Polygon ;
+    using ostk::math::geom::d2::objects::Composite ;
+    using ostk::math::geom::d2::Intersection ;
+
+    class_<Intersection> intersection(aModule, "Intersection") ;
+
+    intersection.def(self == self)
+        .def(self != self)
+
+        .def(self + self)
+        .def(self += self)
+
+        .def("__str__", &(shiftToString<Intersection>))
+        .def("__repr__", &(shiftToString<Intersection>))
+
+        .def("is_defined", &Intersection::isDefined)
+        .def("is_empty", &Intersection::isEmpty)
+        .def("is_complex", &Intersection::isComplex)
+
+        .def("is_point", +[] (const Intersection& anIntersection) -> bool { return anIntersection.is<Point>() ; })
+        .def("is_point_set", +[] (const Intersection& anIntersection) -> bool { return anIntersection.is<PointSet>() ; })
+        .def("is_line", +[] (const Intersection& anIntersection) -> bool { return anIntersection.is<Line>() ; })
+        .def("is_segment", +[] (const Intersection& anIntersection) -> bool { return anIntersection.is<Segment>() ; })
+        .def("is_line_string", +[] (const Intersection& anIntersection) -> bool { return anIntersection.is<LineString>() ; })
+        .def("is_polygon", +[] (const Intersection& anIntersection) -> bool { return anIntersection.is<Polygon>() ; })
+        .def("is_composite", +[] (const Intersection& anIntersection) -> bool { return anIntersection.is<Composite>() ; })
+
+        .def("as_point", +[] (const Intersection& anIntersection) -> Point { return anIntersection.as<Point>() ; })
+        .def("as_point_set", +[] (const Intersection& anIntersection) -> PointSet { return anIntersection.as<PointSet>() ; })
+        .def("as_line", +[] (const Intersection& anIntersection) -> Line { return anIntersection.as<Line>() ; })
+        .def("as_segment", +[] (const Intersection& anIntersection) -> Segment { return anIntersection.as<Segment>() ; })
+        .def("as_line_string", +[] (const Intersection& anIntersection) -> LineString { return anIntersection.as<LineString>() ; })
+        .def("as_polygon", +[] (const Intersection& anIntersection) -> Polygon { return anIntersection.as<Polygon>() ; })
+        .def("as_composite", +[] (const Intersection& anIntersection) -> Composite { return anIntersection.as<Composite>() ; })
+
+        .def("access_composite", &Intersection::accessComposite, return_value_policy::reference)
+
+        .def("get_type", &Intersection::getType)
+
+        // Define static methods
+        .def_static("undefined", &Intersection::Undefined)
+        .def_static("empty", &Intersection::Empty)
+        .def_static("point", &Intersection::Point)
```

Comment:
Any chance you could add the `arg` definitions to the various bindings (this is something we started to do, to be more Python friendly)